tests/ignite/metrics/regression/test_pearson_correlation.py

```diff
@@ -0,0 +1,226 @@
+    c = np.random.randn(4).astype(np.float32)
+    d = np.random.randn(4).astype(np.float32)
+    ground_truth = np.random.randn(4).astype(np.float32)
+
+    m = PearsonCorrelation()
+
+    m.update((torch.from_numpy(a), torch.from_numpy(ground_truth)))
+    np_ans = np_corr(a, ground_truth)
+    assert m.compute() == pytest.approx(np_ans, rel=1e-4)
+
+    m.update((torch.from_numpy(b), torch.from_numpy(ground_truth)))
+    np_ans = np_corr(np.concatenate([a, b]), np.concatenate([ground_truth] * 2))
+    assert m.compute() == pytest.approx(np_ans, rel=1e-4)
+
+    m.update((torch.from_numpy(c), torch.from_numpy(ground_truth)))
+    np_ans = np_corr(np.concatenate([a, b, c]), np.concatenate([ground_truth] * 3))
+    assert m.compute() == pytest.approx(np_ans, rel=1e-4)
+
+    m.update((torch.from_numpy(d), torch.from_numpy(ground_truth)))
+    np_ans = np_corr(np.concatenate([a, b, c, d]), np.concatenate([ground_truth] * 4))
+    assert m.compute() == pytest.approx(np_ans, rel=1e-4)
+
+
+@pytest.fixture(params=list(range(2)))
+def test_case(request):
+    return [
+        (torch.rand(size=(50,)).float(), torch.rand(size=(50,)).float(), 10 ** np.random.uniform(-8, 0), 1),
+        (torch.rand(size=(50, 1)).float(), torch.rand(size=(50, 1)).float(), 10 ** np.random.uniform(-8, 0), 10),
+    ][request.param]
+
+
+@pytest.mark.parametrize("n_times", range(5))
+def test_integration(n_times, test_case: Tuple[Tensor, Tensor, float, int]):
+    y_pred, y, eps, batch_size = test_case
+
+    def update_fn(engine: Engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = np_y[idx : idx + batch_size]
+        y_pred_batch = np_y_pred[idx : idx + batch_size]
+        return torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+
+    engine = Engine(update_fn)
+
+    m = PearsonCorrelation(eps=eps)
+    m.attach(engine, "corr")
+
+    np_y = y.ravel().numpy()
+    np_y_pred = y_pred.ravel().numpy()
+
+    data = list(range(y_pred.shape[0] // batch_size))
+    corr = engine.run(data, max_epochs=1).metrics["corr"]
+
+    np_ans = np_corr(np_y_pred, np_y, eps=eps)
+
+    assert pytest.approx(np_ans, rel=2e-4) == corr
+
+
+def test_accumulator_detached():
+    corr = PearsonCorrelation()
+
+    y_pred = torch.tensor([2.0, 3.0], requires_grad=True)
+    y = torch.tensor([-2.0, -1.0])
+    corr.update((y_pred, y))
+
+    assert all(
+        (not accumulator.requires_grad)
+        for accumulator in (
+            corr._sum_of_products,
+            corr._sum_of_y_pred_squares,
+            corr._sum_of_y_preds,
+            corr._sum_of_y_squares,
+            corr._sum_of_ys,
+        )
+    )
+
+
+@pytest.mark.usefixtures("distributed")
+class TestDistributed:
+    def test_compute(self):
+        rank = idist.get_rank()
+        device = idist.device()
+        metric_devices = [torch.device("cpu")]
+        if device.type != "xla":
+            metric_devices.append(device)
+
+        for i in range(3):
```

Comment:
@kzkadc maybe we can remove this loop and do a single run.
Same below for `test_integration`